src/blog/pointer-compression.md

```diff
@@ -0,0 +1,537 @@
+---
+title: 'Pointer Compression in V8'
+author: 'Igor Sheludko and Santiago Aboy Solanes, *the* pointer compressors'
+avatars:
+  - 'igor-sheludko'
+  - 'santiago-aboy-solanes'
+date: 2020-03-30
+tags:
+  - internals
+  - memory
+description: 'V8 reduced its heap size up to 43%! Learn how in “Pointer Compression in V8”!'
+tweet: ''
+---
+
+There is a constant battle between memory and performance. As users, we would like things to be fast as well as consume as little memory as possible. Unfortunately, usually improving performance comes at a cost of memory consumption (and vice versa).
+
+Back in 2014 Chrome switched from being a 32-bit process to a 64-bit process. This gave Chrome better [security, stability and performance](https://blog.chromium.org/2014/08/64-bits-of-awesome-64-bit-windows_26.html), but it came at a memory cost since each pointer now occupies eight bytes instead of four. We took on the challenge to reduce this overhead in V8 to try and get back as many wasted 4 bytes as possible.
+
+Before diving into the implementation, we need to know where we are standing to correctly assess the situation. To measure our memory and performance we use a set of [web pages](https://v8.dev/blog/optimizing-v8-memory) that reflect popular real-world websites. The data showed that V8 contributes up to 60% of Chrome’s [renderer process](https://www.chromium.org/developers/design-documents/multi-process-architecture) memory consumption on desktop, with an average of 40%.
+
+<figure>
+  <img src="/_img/pointer-compression/memory-chrome.svg" width="600" height="371" alt="V8 memory consumption percentage in Chrome’s renderer memory" loading="lazy">
```

Comment:
Here and on other images: set `alt` to empty with `alt=""`. When we use `figcaption`, there is no point in duplicating same text inside the `alt`, because it will only worsen UX for users on low-speed connections or impaired users, who will now see or hear same text twice.

Reply:
Done.